neo_force_scheme/_neo_force_scheme.py

```diff
@@ -207,27 +208,39 @@ def transform(
             # randomly initialize the projection
             if starting_projection_mode == ProjectionMode.RANDOM:
                 # print(starting_projection_mode)
-                Xd = np.random.random((size, 2))
+                Xd = np.random.random((size, n_dimension))
             # initialize the projection with tsne
             elif starting_projection_mode == ProjectionMode.TSNE:
-                Xd = tsne.excute_tsne(X)
+                Xd = tsne.excute_tsne(X, n_dimension=n_dimension)
             # initialize the projection with pca
             elif starting_projection_mode == ProjectionMode.PCA:
-                Xd = pca.excute_pca(X)
+                Xd = pca.excute_pca(X, n_dimension=n_dimension)
         # create random index TODO: other than random
         index = np.random.permutation(size)
 
         if self.cuda:
             Xd, self.projection_error_ = self._gpu_transform(Xd, index=index, total=total, inplace=inpalce)
         else:
-            Xd, self.projection_error_ = self._transform(Xd, index=index, total=total, inplace=inpalce)
-
-        # setting the min to (0,0)
-        min_x = min(Xd[:, 0])
-        min_y = min(Xd[:, 1])
-        for i in range(size):
-            Xd[i][0] -= min_x
-            Xd[i][1] -= min_y
+            Xd, self.projection_error_ = self._transform(Xd, index=index, total=total, inplace=inpalce,
+                                                         n_dimension=n_dimension)
+
+        if n_dimension == 2:
+            # setting the min to (0,0)
+            min_x = min(Xd[:, 0])
+            min_y = min(Xd[:, 1])
+            for i in range(size):
+                Xd[i][0] -= min_x
+                Xd[i][1] -= min_y
+
+        elif n_dimension == 3:
+            # setting the min to (0,0)
+            min_x = min(Xd[:, 0])
+            min_y = min(Xd[:, 1])
+            min_z = min(Xd[:, 2])
+            for i in range(size):
+                Xd[i][0] -= min_x
+                Xd[i][1] -= min_y
+                Xd[i][2] -= min_z
```

Comment:
Can you update this to be generic in regards with n_dimensions? you can use a for loop for each dimension and store the min_ in a list/array